src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -301,34 +295,42 @@ private static bool CompilationHasOpenFileOnlyAnalyzers(CompilationWithAnalyzers
         {
             try
             {
-                // create result map
                 var version = await GetDiagnosticVersionAsync(project, cancellationToken).ConfigureAwait(false);
 
+                var (fileLoadAnalysisResult, failedDocuments) = await GetDocumentLoadFailuresAsync(project, version, cancellationToken).ConfigureAwait(false);
+                result = result.SetItem(FileContentLoadAnalyzer.Instance, fileLoadAnalysisResult);
+
                 foreach (var analyzer in ideAnalyzers)
                 {
                     var builder = new DiagnosticAnalysisResultBuilder(project, version);
 
-                    if (analyzer is DocumentDiagnosticAnalyzer documentAnalyzer)
+                    switch (analyzer)
                     {
-                        foreach (var document in project.Documents)
-                        {
-                            var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                            if (tree != null)
-                            {
-                                builder.AddSyntaxDiagnostics(tree, await ComputeDocumentDiagnosticAnalyzerDiagnosticsAsync(document, documentAnalyzer, AnalysisKind.Syntax, compilation, cancellationToken).ConfigureAwait(false));
-                                builder.AddSemanticDiagnostics(tree, await ComputeDocumentDiagnosticAnalyzerDiagnosticsAsync(document, documentAnalyzer, AnalysisKind.Semantic, compilation, cancellationToken).ConfigureAwait(false));
-                            }
-                            else
+                        case DocumentDiagnosticAnalyzer documentAnalyzer:
+                            foreach (var document in project.Documents)
                             {
-                                builder.AddExternalSyntaxDiagnostics(document.Id, await ComputeDocumentDiagnosticAnalyzerDiagnosticsAsync(document, documentAnalyzer, AnalysisKind.Syntax, compilation, cancellationToken).ConfigureAwait(false));
-                                builder.AddExternalSemanticDiagnostics(document.Id, await ComputeDocumentDiagnosticAnalyzerDiagnosticsAsync(document, documentAnalyzer, AnalysisKind.Semantic, compilation, cancellationToken).ConfigureAwait(false));
+                                // don't analyze documents whose content failed to load
```

Comment:
Just to be clear - the underlying `Compilation` and `CompilationWithAnalyzers` will still have the syntax tree for failed documents, and the analyzer will execute on these trees. The logic here just filters out the analyzer diagnostics reported on such trees, if any, from the live analysis.

Reply:
Sync'ed offline with @tmat - we are not adding these trees for failed documents to the underlying compilation.

Reply:
Yup, here: https://github.com/dotnet/roslyn/pull/40044/files#diff-4468e5bbe833dfc3d3a7b5699bf1a946R508